### PR TITLE
fix(cli): Conform python and java package names to language standards (no hyphens)

### PR DESCRIPTION
**Pull request by campionfellin**
With new changes in `jsii-srcmak`, it throws an Error when trying to use a package or moduleName with a hyphen (see commit: https://github.com/aws/jsii-srcmak/commit/766de96d655463dadf5a9578ba38d47121be447b)

This is the other side of that and fixes that issue by replacing `-` with `_`.

Tested by changing the `version.yaml` in the python CRD example to use a hyphen in the name. It successfully comes in as an underscore and is able to be imported.

Fixes https://github.com/awslabs/cdk8s/issues/274

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by campionfellin** [comment]
Looks like I missed updating the snapshots in the unit tests. Will do that tomorrow!

File: examples/python/crd/main.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 from constructs import Construct
 from cdk8s import App, Chart
 
-from imports import jenkins
-from imports import clusterinstallation
+from imports.jenkins.io import jenkins
```
Comment (iliapolo) [inline examples/python/crd/main.py:5]: Where did this change come from?

File: packages/cdk8s-cli/src/import/base.ts [hunk 2]
```diff
@@ -69,6 +69,7 @@ export abstract class ImportBase {
 
           const opts: srcmak.Options = {
             entrypoint: fileName,
+            moduleKey: name,
```
Comment (iliapolo) [inline packages/cdk8s-cli/src/import/base.ts:72]: Can you explain?

File: examples/python/crd/main.py (hunk 1 above)
Comment (campionfellin) [reply examples/python/crd/main.py:5]: This was originally noticed in this bug: https://github.com/awslabs/cdk8s/issues/273

The folder structure of the imports is like:

```
▁
/Users/campionfellin/Desktop/issue273/cdk8s/examples/python/crd/imports
├── autoscaling
|  └── internal
|     └── knative
|        └── dev
|           ├── metric
|           └── podautoscaler
├── caching
|  └── internal
|     └── knative
|        └── dev
|           └── image
├── jenkins
|  └── io
|     └── jenkins
|        ├── __init__.py
|        ├── _jsii
|        |  ├── __init__.py
|        |  └── jenkinsiojenkins@0.0.0.jsii.tgz
|        └── py.typed
├── k8s
|  ├── __init__.py
|  ├── _jsii
|  |  ├── __init__.py
|  |  └── k8s@0.0.0.jsii.tgz
|  └── py.typed
├── mattermost
|  └── com
|     └── clusterinstallation
|        ├── __init__.py
|        ├── _jsii
|        |  ├── __init__.py
|        |  └── mattermostcomclusterinstallation@0.0.0.jsii.tgz
|        └── py.typed
├── networking
|  └── internal
|     └── knative
|        └── dev
|           ├── certificate
|           ├── ingress
|           └── serverlessservice
├── sample_controller
|  └── k8s
|     └── io
|        └── foo
|           ├── __init__.py
|           ├── _jsii
|           └── py.typed
└── serving
   └── knative
      └── dev
         ├── configuration
         |  ├── __init__.py
         |  ├── _jsii
         |  └── py.typed
         ├── revision
         |  ├── __init__.py
         |  ├── _jsii
         |  └── py.typed
         ├── route
         |  ├── __init__.py
         |  ├── _jsii
         |  └── py.typed
         └── service
            ├── __init__.py
            ├── _jsii
            └── py.typed
```

So the import as it was, failed with:

```
    from imports import clusterinstallation
ImportError: cannot import name 'clusterinstallation' from 'imports' (unknown location)
```

File: packages/cdk8s-cli/src/import/base.ts (hunk 2 above)
Comment (campionfellin) [reply packages/cdk8s-cli/src/import/base.ts:72]: The `moduleKey` for `jsii-srcmak` is so consumers can choose how to set the package name in the `package.json` before running it through JSII (https://github.com/aws/jsii-srcmak/blob/master/src/compile.ts#L55)

This was previously just `generated` but there was a problem with multiple imports colliding with the same package name. There's a bunch more details in https://github.com/awslabs/cdk8s/issues/273 as well!

Comment (iliapolo) [reply packages/cdk8s-cli/src/import/base.ts:72]: Gotcha 👍

**Pull request merged**